src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1587,11 +1583,16 @@ Connection connectInternal(Properties propsIn,
             }
 
             if (authenticationString.equalsIgnoreCase(SqlAuthentication.NTLM.toString())) {
+                String sPropKeyDomain = SQLServerDriverStringProperty.DOMAIN_NAME.toString();
+                String sPropValueDomain = activeConnectionProperties.getProperty(sPropKeyDomain);
+                String domainName = sPropValueDomain;
+
                 // domain and no user or password
-                if (!StringUtils.isEmpty(domainName) && ((activeConnectionProperties
-                        .getProperty(SQLServerDriverStringProperty.USER.toString()).isEmpty())
-                        || (activeConnectionProperties.getProperty(SQLServerDriverStringProperty.PASSWORD.toString())
-                                .isEmpty()))) {
+                if (null == domainName || domainName.isEmpty()
+                        || !domainName.isEmpty() && ((activeConnectionProperties
```

Comment:
You need an opening bracket before !domainName.isEmpty() ending at the end of condition..